cfme/fixtures/v2v_fixtures.py

```diff
@@ -0,0 +1,449 @@
+    """Return Infra Mapping form data and vm object"""
+    infra_mapping_data = infra_mapping_default_data(source_provider, provider)
+    recursive_update(
+        infra_mapping_data,
+        {
+            "description": "Single DS migration of VM from {ds_type1} to {ds_type2},".format(
+                ds_type1=request.param[0], ds_type2=request.param[1]
+            ),
+            "datastores": [
+                component_generator(
+                    "datastores", source_provider, provider, request.param[0], request.param[1]
+                )
+            ],
+        },
+    )
+    vm_obj = get_vm(request, appliance, source_provider, request.param[2], request.param[0])
+    return FormDataVmObj(infra_mapping_data=infra_mapping_data, vm_list=[vm_obj])
+
+
+@pytest.fixture(scope="function")
+def mapping_data_vm_obj_single_network(request, appliance, source_provider, provider):
+    infra_mapping_data = infra_mapping_default_data(source_provider, provider)
+    recursive_update(
+        infra_mapping_data,
+        {
+            "description": "Single Network migration of VM from {vlan1} to {vlan2},".format(
+                vlan1=request.param[0], vlan2=request.param[1]
+            ),
+            "networks": [
+                component_generator(
+                    "vlans", source_provider, provider, request.param[0], request.param[1]
+                )
+            ],
+        },
+    )
+    vm_obj = get_vm(request, appliance, source_provider, request.param[2])
+    return FormDataVmObj(infra_mapping_data=infra_mapping_data, vm_list=[vm_obj])
+
+
+def component_generator(selector, source_provider, provider, source_type=None, target_type=None):
+    """
+    Component generator method to generate a dict of source and target
+    components(clusters/datastores/networks).
+    Gets the provider data based on selector from cfme_data.yaml and creates
+    InfraMapping.component(source_list, target_list) object
+
+    Test is skipped if no source or target data is found
+
+    Args:
+        selector: can be clusters/datastores/networks
+        source_provider: vmware provider to migrate from
+        provider:  rhev or osp provider or target provider to migrate to
+        source_type: string source datastores/networks on vmware provider to migrate from.
+                     Ex: if source_type is "iscsi". Provider data is checked for datastore with type
+                     iscsi and that datastores name is used.
+        target_type: string target datastores/networks to migrate to
+
+        returns : InfraMapping.component(source_list, target_list) object
+
+    """
+    if selector not in "clusters" or "datastores" or "vlans":
+        raise ValueError("Please specify cluster, datatsore or network(vlans) selector!")
+
+    source_data = source_provider.data.get(selector, [])
+    target_data = provider.data.get(selector, [])
+    component = None
+    if selector is "clusters":
```

Comment:
I still think the logic is a bit broken here

If `source_data` was empty, `sources[0]` will raise TypeError because `sources` is now `None` and has no `__get__` method.

You only want to look at the selector if `source_data` and `target_data` are not empty. You can call `pytest.skip` immediately, or move the selector checks into an `if` block, so that when they're empty it skips straight to the existing check+skip.